pkg/kubectl/cmd/delete.go

```diff
@@ -251,7 +251,8 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 		if err != nil {
 			// If there is no reaper for this resources and the user didn't explicitly ask for stop.
 			if kubectl.IsNoSuchReaperError(err) && isDefaultDelete {
-				return deleteResource(info, out, shortOutput, mapper)
+				// No client side reaper found. Let the server do cascading deletion.
+				return cascadingDeleteResource(info, out, shortOutput, mapper)
```

Comment:
> Is it only the federation resources that are missing reapers?

Kubernetes resources are also missing reapers for resources where there is nothing to reap (for ex: secrets, configmaps). In federation, we need to reap them from underlying clusters. This change should not have any impact on any kubernetes resource.

> Is it possible that users are relying on kubectl not cascading delete federation resources?

Cascading deletion should happen when users use `kubectl delete --cascading=true` with federation. Right now, it cascading deletes some resources (like namespaces, rs and deployments) and does not cascading deletes others resources (like secrets, configmaps). This is a bug that this PR fixes.

Default behavior is being discussed in https://github.com/kubernetes/kubernetes/issues/38897. The consensus seems to be to be consistent with kubernetes (i.e cascading delete by default).